BrainPortal/script/show_cbrain_rev

```diff
@@ -34,26 +34,11 @@ end
 
 # Code lifted almost as-is from cbrain_file_revision.rb
 
-    git_tag = nil
-    seen    = {}
     Dir.chdir(top) do
-      tags_set = `git tag -l`.split.shuffle # initial list: all tags we can find
-      git_tag = tags_set.shift unless tags_set.empty? # extract one as a starting point
-      seen[git_tag]=true
-      while tags_set.size > 0
-        tags_set = `git tag --contains '#{git_tag}'`.split.shuffle.reject { |v| seen[v] }
-        git_tag = tags_set.shift unless tags_set.empty? # new first
-        seen[git_tag]=true
-      end
-      if git_tag
-        num_new_commits = `git rev-list '#{git_tag}..HEAD'`.split.size
-      elsif default_tag # fallback for plugins, when -z used
-        num_new_commits = `git rev-list HEAD`.split.size
-        git_tag = default_tag
-      end
-      if git_tag
-        git_tag += "-#{num_new_commits}" if num_new_commits > 0
-        puts git_tag
+      out = `git describe --tags --long`
```

Comment:
Same here.. Al of this could be

```
  `git describe --tags --long`
  .strip
  .sub(/(-0)?-\w+$/,"")
```

Reply:
thanks, but can replace if you prefer. I know that regex is powerful and concise, yet I believe tat re is a bit condense for most people, me included, so we have more trouble with deciphering regexes than with string operations involving partition/delete_suffix/split and alike, so tried ditch re.

Reply:
That's a terrible argument. This regex is very simple. If you think it's still too complex then you write it with the extended syntax that allows you to embed comments:

```
  `git describe --tags --long`
  .strip
  .sub(
    /
      (-0)?           # optional dash zero, like "-0"
      -\w+            # a dash and alphanums like "-b027af3"
     $                  # anchor at end of string
    /x ,
    ""                  # replace with nothing
 )
```

To me that is overkill, and the straight version ` .sub( /(-0)?-\w+$/,  "" ) ` is totally readable.

Reply:
well, IMHO. stackoverflow is filled with regex hacks question. I shortened the old version a bit, hope better now